src/python/WMCore/MicroService/MSPileup/MSPileupData.py

```diff
@@ -0,0 +1,215 @@
+"""
+File       : MSPileupData.py
+Author     : Valentin Kuznetsov <vkuznet AT gmail dot com>
+Description: MSPileupData provides logic behind
+data used and stored by MSPileup module.
+
+The data flow should be done via list of objects, e.g.
+[<pileup object 1>, <pileup object 2>, ..., <pileup object n>]
+"""
+
+# system modules
+import time
+import calendar
+
+# third party modules
+from pymongo import IndexModel, errors
+
+# WMCore modules
+from WMCore.Database.MongoDB import MongoDB
+from WMCore.MicroService.MSPileup.DataStructs.MSPileupObj import MSPileupObj, gmtimeSeconds
+from WMCore.MicroService.MSPileup.MSPileupError import MSPileupUniqueConstrainError, \
+    MSPileupDatabaseError, MSPileupNoKeyFoundError, MSPileupDuplicateDocumentError
+from WMCore.MicroService.Tools.Common import getMSLogger
+
+
+def stripKeys(docs, skeys=None):
+    """
+    Helper function to strip out keys from given dictionary.
+
+    :param docs: either input dictionary or list of dictionaries
+    :param skeys: list of of strip keys
+    """
+    if skeys is None:
+        return docs
+    if isinstance(docs, list):
+        results = []
+        for doc in docs:
+            if doc:
+                for key in skeys:
+                    doc.pop(key, None)
+                    results.append(doc)
+        return results
+
+    if docs and isinstance(docs, dict):
+        for key in skeys:
+            docs.pop(key, None)
+    return docs
+
+
+def encodeTimestamps(doc):
+    """
+    Encode timestamps in MSPileup data document
+    """
+    for key in ["insertTime", "lastUpdateTime", "activatedOn", "deactivatedOn"]:
+        val = doc.get(key, None)
+        if val and isinstance(val, int):
+            # convert to string representation
+            doc[key] = time.strftime("%Y-%m-%dT%H:%M:%SZ", time.gmtime(val))
+    return doc
+
+
+def decodeTimestamps(doc):
```

Comment:
Same suggestion as made above for the `encodeTimestamps` function.

Reply:
done